Index: lib/language/qlanguagefactory.h
===================================================================
--- lib/language/qlanguagefactory.h	(révision 758)
+++ lib/language/qlanguagefactory.h	(copie de travail)
@@ -54,6 +54,9 @@
 		
 		virtual QLanguageDefinition* definition(const QString& file);
 		virtual QCodeCompletionEngine* completionEngine(const QString& file);
+
+		const QList<QCodeCompletionEngine*> & engines() const;
+		const QList<QLanguageDefinition*> & languageDefinitions() const;
 		
 		inline QFormatFactory* formatFactory() const { return pFormatFactory; }
 		
Index: lib/language/qlanguagefactory.cpp
===================================================================
--- lib/language/qlanguagefactory.cpp	(révision 758)
+++ lib/language/qlanguagefactory.cpp	(copie de travail)
@@ -332,4 +332,12 @@
 	}
 }
 
+const QList<QCodeCompletionEngine*> & QLanguageFactory::engines() const {
+	return m_engines;
+}
+
+const QList<QLanguageDefinition*> & QLanguageFactory::languageDefinitions() const {
+	return definitions;
+}
+
 /*! @} */
