Index: build-mingw.bat
===================================================================
--- build-mingw.bat	(revision 760)
+++ build-mingw.bat	(working copy)
@@ -5,9 +5,9 @@
 :: build dbus sources
 :: 
 
-set CMAKE=\cmake-2.4.4-win32-x86\bin\cmake.exe
-set MINGW=\mingw
-set PATH=%MINGW%\bin;%ProgramFILES%\gnuwin32\bin;%PATH%
+set CMAKE=\Qt\CMake\bin\cmake.exe
+set MINGW=\Qt\mingw
+set PATH=%MINGW%\bin;C:\Qt\gnuwin32\bin;C:\Qt\win32libs\bin;%PATH%
 set SOURCE_DIR=%CD%
 
 if not exist ..\windbus-build (
@@ -18,7 +18,7 @@
 )
 
 cd ..\windbus-build
-%CMAKE% -G "MinGW Makefiles" %SOURCE_DIR%\cmake
+%CMAKE% -G "MinGW Makefiles" %SOURCE_DIR%\cmake -DDBUS_USE_EXPAT=ON -DLIBEXPAT_INCLUDE_DIR=C:\Qt\win32libs\include -DDBUS_INSTALL_SYSTEM_LIBS=1
 mingw32-make
 bin\dbus-test test\data
 ::bin\bus-test test\data
Index: bus/main.c
===================================================================
--- bus/main.c	(revision 760)
+++ bus/main.c	(working copy)
@@ -241,9 +241,12 @@
     reload_pipe[RELOAD_WRITE_END] = -1;
 }
 
+#include <windows.h>
+
 int
 main (int argc, char **argv)
 {
+  FreeConsole();
   DBusError error;
   DBusString config_file;
   DBusString addr_fd;
