Index: build-mingw.bat
===================================================================
--- build-mingw.bat	(revision 760)
+++ build-mingw.bat	(working copy)
@@ -5,9 +5,9 @@
 :: build dbus sources
 :: 
 
-set CMAKE=\cmake-2.4.4-win32-x86\bin\cmake.exe
-set MINGW=\mingw
-set PATH=%MINGW%\bin;%ProgramFILES%\gnuwin32\bin;%PATH%
+set CMAKE=\usr_win32\cmake\bin\cmake.exe
+set MINGW=\usr_win32\MinGW\bin
+set PATH=%MINGW%\bin;\usr_win32\win32libs\bin;%PATH%
 set SOURCE_DIR=%CD%
 
 if not exist ..\windbus-build (
@@ -18,7 +18,7 @@
 )
 
 cd ..\windbus-build
-%CMAKE% -G "MinGW Makefiles" %SOURCE_DIR%\cmake
+%CMAKE% -G "MinGW Makefiles" %SOURCE_DIR%\cmake -DDBUS_USE_EXPAT=ON -DLIBEXPAT_INCLUDE_DIR=C:\Qt\win32libs\include -DDBUS_INSTALL_SYSTEM_LIBS=1
 mingw32-make
 bin\dbus-test test\data
 ::bin\bus-test test\data
Index: cmake/bus/CMakeLists.txt
===================================================================
--- cmake/bus/CMakeLists.txt	(revision 760)
+++ cmake/bus/CMakeLists.txt	(working copy)
@@ -80,10 +80,10 @@
 
 if(splitlib AND MSVC)
 	add_library(bus-lib STATIC ${BUS_SOURCES})
-	add_executable(dbus-daemon${CMAKE_EXE_POSTFIX} ${BUS_DIR}/main.c)
+	add_executable(dbus-daemon${CMAKE_EXE_POSTFIX} WIN32 ${BUS_DIR}/main.c)
 	target_link_libraries(dbus-daemon${CMAKE_EXE_POSTFIX} ${DBUS_1} ${LIBS} bus-lib)
 else(splitlib AND MSVC)
-	add_executable(dbus-daemon${CMAKE_EXE_POSTFIX} ${BUS_SOURCES} ${BUS_DIR}/main.c)
+	add_executable(dbus-daemon${CMAKE_EXE_POSTFIX} WIN32 ${BUS_SOURCES} ${BUS_DIR}/main.c)
 	target_link_libraries(dbus-daemon${CMAKE_EXE_POSTFIX} ${DBUS_1} ${LIBS})
 endif(splitlib AND MSVC)
 
